src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,97 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size = 0;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        int index = getIndex(key);
+        Node<K, V> newNode = new Node<>(key, value);
+
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+        } else {
+            Node<K, V> current = table[index];
+            while (true) {
+                if (Objects.equals(current.key, key)) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = newNode;
+                    size++;
+                    break;
+                }
+                current = current.next;
+            }
+        }
 
+        if (size >= DEFAULT_LOAD_FACTOR * table.length) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> current = table[index];
+
+        while (current != null) {
+            if (Objects.equals(current.key, key)) {
+                return current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        return (key == null) ? 0 : Math.abs(Objects.hashCode(key) % table.length);
+    }
+
+    public void resize() {
+        int newCapacity = table.length * 2;
+        Node<K, V>[] newTable = new Node[newCapacity];
+
+        for (int i = 0; i < table.length; i++) {
+            Node<K, V> current = table[i];
+            while (current != null) {
+                Node<K, V> next = current.next;
+
+                int newIndex = Math.abs(current.key.hashCode() % newCapacity);
+
+                current.next = newTable[newIndex];
+                newTable[newIndex] = current;
+
+                current = next;
+            }
+        }
+        table = newTable;
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
```

Comment:
Consider initializing the `next` pointer in the `Node` constructor to `null` explicitly. This can help avoid potential `NullPointerException` when traversing nodes.